x-pack/legacy/plugins/ml/server/models/job_service/new_job/population_chart.ts

```diff
@@ -0,0 +1,266 @@
+        // TODO event rate
+        // populationBuckets.forEach(b => {
+        //   // check to see if the data is split.
+        //   if (b[i] === undefined) {
+        //     values.push({ label: b.key, value: b.doc_count });
+        //   } else {
+        //     // a split is being used, so an additional filter was added to the search
+        //     values.push({ label: b.key, value: b[i].doc_count });
+        //   }
+        // });
+      } else if (typeof dataForTime.population !== 'undefined') {
+        populationBuckets.forEach((b: any) => {
+          const tempBucket = b[i];
+          let value = null;
+          // check to see if the data is split
+          // if the field has been split, an additional filter and aggregation
+          // has been added to the search in the form of splitValue
+          const tempValue =
+            tempBucket.value === undefined && tempBucket.splitValue !== undefined
+              ? tempBucket.splitValue
+              : tempBucket;
+
+          // check to see if values is exists rather than value.
+          // if values exists, the aggregation was median
+          if (tempValue.value === undefined && tempValue.values !== undefined) {
+            value = tempValue.values[ML_MEDIAN_PERCENTS];
+          } else {
+            value = tempValue.value;
+          }
+          values.push({ label: b.key, value: isFinite(value) ? value : null });
+        });
+      }
+
+      // const highestValueField = _.reduce(values, (p, c) => (c.value > p.value ? c : p), {
+      //   value: 0,
+      // });
+
+      tempResults[i].push({
+        time,
+        values,
+      });
+
+      // if (this.chartData.detectors[i]) {
+      //   this.chartData.detectors[i].line.push({
+      //     date,
+      //     time,
+      //     values,
+      //   });
+
+      // init swimlane
+      // this.chartData.detectors[i].swimlane.push({
+      //   date,
+      //   time,
+      //   value: 0,
+      //   color: '',
+      //   percentComplete: 0,
+      // });
+
+      // this.chartData.detectors[i].highestValue = Math.ceil(
+      //   Math.max(this.chartData.detectors[i].highestValue, Math.abs(highestValueField.value))
+      // );
+      // }
+
+      // tempResults[i].push({
+      //   time,
+      //   value,
+      // });
+    });
+  });
+
+  // const results: Record<number, Result[]> = {};
+  // Object.entries(tempResults).forEach(([fieldIdx, results2]) => {
+  //   results[+fieldIdx] = results2.sort((a, b) => a.time - b.time);
+  // });
+
+  return {
+    success: true,
+    results: tempResults,
+    totalResults: resp.hits.total,
+  };
+}
+
+function getPopulationSearchJsonFromConfig(
+  indexPatternTitle: string,
+  timeField: string,
+  start: number,
+  end: number,
+  intervalMs: number,
+  query: any,
+  aggFieldNamePairs: AggFieldNamePair[],
+  splitFieldName: string | null
+): object {
+  const json = {
+    index: indexPatternTitle,
+    size: 0,
+    rest_total_hits_as_int: true,
```

Comment:
Is `hits.total` being used from the results for this search? If not, this shouldn't be needed.

Reply:
`resp.hits.total` is being used as `totalResults` a few lines above